tests/func/test_gc.py

```diff
@@ -327,3 +328,16 @@ def test_gc_rev_num(tmp_dir, scm, dvc):
             assert not cache.exists()
         else:
             assert cache.read_text() == str(i)
+
+
+def test_date(tmp_dir, scm, dvc):
+    tmp_dir.dvc_gen("testfile", "content", commit="add testfile")
+
+    now = datetime.datetime.now()
+    datestamp = now.date().replace(day=now.day + 1).isoformat()
```

Comment:
Can we add comparison tests? for example `cmoit_date=now.day - 1` and no cache deleted?

Reply:
Do you mean something like this?

```diff
diff --git a/tests/func/test_gc.py b/tests/func/test_gc.py
index 469230f4..2e8acd77 100644
--- a/tests/func/test_gc.py
+++ b/tests/func/test_gc.py
@@ -341,3 +341,9 @@ def test_date(tmp_dir, scm, dvc):
     dvc.gc(commit_date=datestamp)
 
     assert _count_files(dvc.odb.local.path) == 1
+
+    tmp_dir.dvc_gen("testfile", "modified, again", commit="modify")
+
+    datestamp = now.date().replace(day=now.day - 1).isoformat()
+    dvc.gc(commit_date=datestamp)
+    assert _count_files(dvc.odb.local.path) == 2

```